src/MetadataManager/metadataAPI.ts

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+
+import {ToolchainInfo} from '../Backend/Toolchain';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+
+import {PathToHash} from './pathToHash';
+
+interface RelationInfo {
+  'selected': string, 'relation-data': Node[]
+}
+
+interface Node {
+  'id': string, 'parent': string, 'represent-idx': number, 'data-list': Data[]
+}
+
+interface Data {
+  'path': string, 'name': string, 'onecc-version'?: string, 'toolchain-version'?: string,
+      'is-deleted': boolean
+}
+
+type BuildInfoKeys = 'onecc'|'toolchain'|'cfg';
+
+interface BuildInfo {
+  'onecc': string, 'toolchain': ToolchainInfo|undefined, 'cfg': any
+}
+
+export class Metadata {
+  constructor() {}
+  private static _buildInfoMap = new Map<string, BuildInfo>();
+  private static _childToParentMap = new Map<string, string>();
+
+  public static setRelationMap(path: string, parentPath: string) {
+    this._childToParentMap.set(path, parentPath);
+  }
+
+  public static async setRelation(uri: vscode.Uri) {
```

Comment:
It seems some logic can be separated into 'Relation', 'Metadata', 'Hash' or so on. Please group those functions into related ones and try separating some as a module, so that we can reduce dependencies between 'Metadata' and 'PathToHash' modules.